mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -0,0 +1,736 @@
+            uuid,
+            key_trade_pair.as_combination()
+        ),
+    };
+}
+
+pub async fn lp_bot_loop(ctx: MmArc) {
+    info!("lp_bot_loop successfully started");
+    loop {
+        // todo: this log should probably in debug
+        info!("tick lp_bot_loop");
+        if ctx.is_stopping() {
+            // todo: can we cancel all the pending orders when the ctx is stopping or call tear_down ?
+            break;
+        }
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+        let mut states = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        if *states == TradingBotState::Stopping {
+            *states = TradingBotState::Stopped;
+            // todo: verify if there is a possible deadlock here if i use states inside tear_down_bot
+            tear_down_bot(ctx).await;
+            break;
+        }
+        drop(states);
+        process_bot_logic(&ctx).await;
+        Timer::sleep(30.0).await;
+    }
+    info!("lp_bot_loop successfully stopped");
+}
+
+pub async fn process_price_request() -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    info!("Fetching price from: {}", KMD_PRICE_ENDPOINT);
+    let (status, headers, body) = slurp_url(KMD_PRICE_ENDPOINT).await?;
+    let (status_code, body, _) = (status, std::str::from_utf8(&body)?.trim().into(), headers);
+    if status_code != StatusCode::OK {
+        return MmError::err(PriceServiceRequestError::HttpProcessError(body));
+    }
+    let model: HashMap<String, TickerInfos> = serde_json::from_str(&body)?;
+    Ok(TickerInfosRegistry(model))
+}
+
+async fn fetch_price_tickers() -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    let model = process_price_request().await?;
+    info!("price registry size: {}", model.0.len());
+    Ok(model)
+}
+
+pub async fn start_simple_market_maker_bot(ctx: MmArc, req: StartSimpleMakerBotRequest) -> StartSimpleMakerBotResult {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+    {
+        let mut state = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        match *state {
+            TradingBotState::Running => return MmError::err(StartSimpleMakerBotError::AlreadyStarted),
+            TradingBotState::Stopping => return MmError::err(StartSimpleMakerBotError::CannotStartFromStopping),
+            TradingBotState::Stopped => *state = TradingBotState::Running,
+        }
+        let mut trading_bot_cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await;
+        *trading_bot_cfg = req.cfg;
+    }
+
+    info!("simple_market_maker_bot successfully started");
+    spawn(lp_bot_loop(ctx.clone()));
+    Ok(StartSimpleMakerBotRes {
+        result: "Success".to_string(),
+    })
+}
+
+pub async fn stop_simple_market_maker_bot(ctx: MmArc, _req: Json) -> StopSimpleMakerBotResult {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+    {
```

Comment:
This can be simplified to
```rust
        let mut state = simple_market_maker_bot_ctx.trading_bot_states.lock().await;

        match *state {
            TradingBotState::Stopped =>MmError::err(StopSimpleMakerBotError::AlreadyStopped),
            TradingBotState::Stopping => MmError::err(StopSimpleMakerBotError::AlreadyStopping),
            TradingBotState::Running => {
                    *state = TradingBotState::Stopping;
                    info!("simple_market_maker_bot will stop within 30 seconds");
                    Ok(StopSimpleMakerBotRes {
                          result: "Success".to_string(),
                    })
               }
        }
```
Please note that I changed `_` to the explicit `TradingBotState::Running`. If we add more states, we can occasionally forget to update this fn if they won't be valid for stopping.